cdci_data_analysis/analysis/instrument.py

```diff
@@ -716,6 +731,33 @@ def set_input_products(self, par_dic, input_file_path,input_prod_list_name):
 
         return has_prods
 
+    def upload_files_request(self, par_dic, request, upload_dir, products_url, token=None):
+        list_uploaded_files = []
+        if request.method == 'POST':
+            if validators.url(products_url):
+                basepath = os.path.join(products_url, 'dispatch-data/download_file')
+            else:
+                basepath = os.path.join(products_url, 'download_file')
+            for f in request.files:
+                # TODO needed since those two files are extracted in a previous step
+                if f != 'user_scw_list_file' and f != 'user_catalog_file':
+                    f_path = upload_file(f, upload_dir)
+                    if f_path is not None:
+                        file_hash = make_hash_file(f_path)
+                        f_name, f_ext = os.path.splitext(os.path.basename(f_path))
+                        new_file_name = f_name.split('.')[0] + '_' + file_hash + f_ext
```

Comment:
> Just keeping hash here is better. The request must result in the same `jobid` (and it will depend on the constructed url string, right?) for the same file regardless of its name. 

Yes, the file upload, which happens one line above, uses the name of the field in the request files, to which the hash will be added

>  The metatadata file can then be used to relate the hash to human-readable filename.

It is actually used to relate a file to the users that can access it

Reply:
so, are you suggesting of not using the file_name, and just its hash, for storing the file itself?